### PR TITLE
Cleanup in utils

### PR DESCRIPTION
**Pull request by TomFischer**
- Use `FileIO::writeGeometryToFile()`
- Use `LogogSetup`
- Use `std::unique_ptr`

### CODE REVIEW

**Comment by bilke** [comment]
:+1:

File: Applications/Utils/FileConverter/generateMatPropsFromMatID.cpp [hunk 1]
```diff
@@ -51,7 +49,11 @@ int main (int argc, char* argv[])
 	cmd.parse( argc, argv );
 
 	// read mesh
-	MeshLib::Mesh* mesh(FileIO::readMeshFromFile(mesh_arg.getValue()));
+	std::unique_ptr<MeshLib::Mesh> mesh(FileIO::readMeshFromFile(mesh_arg.getValue()));
+	if (mesh.get() == nullptr) {
```
Comment (chleh) [inline Applications/Utils/FileConverter/generateMatPropsFromMatID.cpp]: `if(mesh)` would be possible, but I do not insist.

File: Applications/Utils/FileConverter/generateMatPropsFromMatID.cpp [hunk 2]
```diff
@@ -81,10 +86,9 @@ int main (int argc, char* argv[])
 	// set mat ids to 0 and write new msh file
 	std::fill(materialIds->begin(), materialIds->end(), 0);
 
-	std::string new_mshname(name + "_new.vtu");
+	std::string const new_mshname(name + "_new.vtu");
 	INFO("Writing mesh to file \"%s\".", new_mshname.c_str());
-	FileIO::VtuInterface mesh_io(mesh);
-	mesh_io.writeToFile (new_mshname);
+	FileIO::writeMeshToFile(*(mesh.get()), new_mshname);
```
Comment (chleh) [inline Applications/Utils/FileConverter/generateMatPropsFromMatID.cpp]: Only as a note: `*mesh` would be enough.

File: Applications/Utils/SimpleMeshCreation/createMeshElemPropertiesFromASCRaster.cpp [hunk 3]
```diff
@@ -252,19 +241,12 @@ int main (int argc, char* argv[])
 			(*materialIds)[dest_mesh->getElement(dest_perm[k])->getID()] = k;
 		}
 
-		FileIO::Legacy::MeshIO mesh_writer;
-		mesh_writer.setPrecision(12);
-		mesh_writer.setMesh(dest_mesh);
-		mesh_writer.writeToFile(out_mesh_arg.getValue());
+		FileIO::writeMeshToFile(*dest_mesh, out_mesh_arg.getValue());
```
Comment (chleh) [inline Applications/Utils/SimpleMeshCreation/createMeshElemPropertiesFromASCRaster.cpp]: You could transform `dest_mesh` into a `unique_ptr`, too, like you did in the file above.

File: Applications/Utils/SimpleMeshCreation/generateStructuredMesh.cpp [hunk 4]
```diff
@@ -217,9 +217,7 @@ int main (int argc, char* argv[])
 		INFO("Mesh created: %d nodes, %d elements.", mesh->getNNodes(), mesh->getNElements());
 
 		// write into a file
-		FileIO::Legacy::MeshIO meshIO;
-		meshIO.setMesh(mesh);
-		meshIO.writeToFile(mesh_out.getValue());
+		FileIO::writeMeshToFile(*mesh, mesh_out.getValue());
```
Comment (chleh) [inline Applications/Utils/SimpleMeshCreation/generateStructuredMesh.cpp]: Same as above.

**Comment by chleh** [comment]
Minor comments, do changes if you want.
:+1:

File: Applications/Utils/FileConverter/generateMatPropsFromMatID.cpp [hunk 5]
```diff
@@ -81,10 +86,9 @@ int main (int argc, char* argv[])
 	// set mat ids to 0 and write new msh file
 	std::fill(materialIds->begin(), materialIds->end(), 0);
```
Comment (rinkk) [inline Applications/Utils/FileConverter/generateMatPropsFromMatID.cpp]: Now that we have property vectors it does not make sense to keep mat ids. Setting all of them to 0 was the next best thing when actually deleting mat ids was not possible. The vector should be removed now.

**Comment by endJunction** [comment]
@TomFischer see PR to your repo...

**Pull request merged**